doc/amazon_sagemaker_debugger.rst

```diff
@@ -0,0 +1,377 @@
+
+**Example 3**: Using a built-in rule with no customization and another built-in rule with customization.
+
+Here we modify the ``weight_update_ratio`` rule to store a custom collection rather than "weights" which it would normally do if the behavior is not overridden.
+
+
+.. code:: python
+
+    from sagemaker.debugger import Rule
+    from smdebug_rulesconfig import vanishing_gradient, weight_update_ratio
+
+    wur_with_customization = Rule.sagemaker(
+        base_config=weight_update_ratio(),
+        name="custom_wup_rule_name",
+        rule_parameters={
+            'key1': 'value1',
+            'key2': 'value2'
+        },
+        collections_to_save=[
+            CollectionConfig(
+                name="custom_collection_name",
+                parameters= {
+                    'key1': 'value1',
+                    'key2': 'value2'
+                }
+            )
+        ]
+    )
+
+    estimator = TensorFlow(
+            role=role,
+            train_instance_count=1,
+            train_instance_type=train_instance_type,
+            rules=[
+                Rule.sagemaker(vanishing_gradient()),
+                wur_with_customization
+            ]
+    )
+
+
+In the example above, collection configuration for Vanishing Gradient is pulled from `SageMaker Debugger Rules Config <https://github.com/awslabs/sagemaker-debugger-rulesconfig>`__  and the user supplied configuration is used for the Weight Update Ratio rule.
+
+Using custom rules
+------------------
+
+SageMaker Debugger also allows the users to create custom rules and have those evaluated against the debugging data. To use custom rules there are two prerequisites:
+
+* Custom rule source file and its local or S3 location. You can learn more about how to write custom rules at `How to Write Custom Debugger Rules <https://github.com/awslabs/sagemaker-debugger/blob/master/docs/analysis.md#writing-a-custom-rule>`__
+* Rule evaluator image for the corresponding region available from `Amazon SageMaker Debugger Custom Rule Images <https://docs.aws.amazon.com/sagemaker/latest/dg/debuger-custom-rule-registry-ids.html>`__
+
+To learn more about how to write your custom rules and use them see `SageMaker Debugger Custom Rules <https://docs.aws.amazon.com/sagemaker/latest/dg/debugger-custom-rules.html>`__
+
+Sample Usage
+~~~~~~~~~~~~
+
+For this example, we evaluate an altered version of the Vanishing Gradient rule against our model training. The rule checks the gradients and asserts that the mean value of the gradients at any step is always above a certain threshold. The source code for the rule is available `here <https://github.com/awslabs/amazon-sagemaker-examples/blob/master/sagemaker-debugger/tensorflow_keras_custom_rule/rules/my_custom_rule.py>`__ and is assumed to be in the relative directory path ``rules/custom_gradient_rule.py``. The region where you'll run the training job is assumed to be ``us-east-1``.
```

Comment:
The last sentence could be misinterpreted to mean that this has to be run in us-east-1, when the only part of the code that is region-specific is the image URI. instead, I'd remove the sentence, and then in the code do something like:

```python
region = 'us-east-1'  # the AWS region of the training job
image_uri = '864354269164.dkr.ecr.{}.amazonaws.com/sagemaker-debugger-rule-evaluator:latest'.format(region)
```

Reply:
Good suggestion. Done.